app/controllers/products_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ProductsController < ApplicationController
+  def new
+    # give them a shell and invite them to fill out the data.  Allows introspection into the object in the view!
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.new(product_create_params[:product])
+    #any validation?
+    if @product.save
+      redirect_to "/vendors/#{@product.vendor_id}/products"
```

Comment:
You can also pass in a vendor_id without string interpolation like so: 
redirect_to vendor_products_path(@product.vendor_id) 
When you rake routes in your terminal you can see the 'shorthand' path name that Rails creates for each of your routes (i.e. root_path, vendors_path, etc.)